examples/soft-kmeans/README.md

```diff
@@ -0,0 +1,108 @@
+- **Components weights:** Since the clusters do not contain the same number of points, weights were set on the components of the mixture at `w1 = 0.4` and `w2 = 0.6` (which fits the proportions of data points in each cluster).
+- **Parameters to fit:** To keep things simple, only the means of the components were set a variable parameters. The standard deviations were fixed at `sigma = 1` (which fits the data generation model), and components weights were also kept fixed.
+- **Prior parameters:** The prior parameters for both components of the gaussian mixture were arbitrarily set at mean `mu=0` and standard deviation `sigma = 5`.
+
+The log-posterior distribution is visualized below:
+
+<img alt="posterior" src="./images/posterior_1D.png" height="400"/>
+
+The posterior shows two modes, which correspond to the two possible positions of the mixture model to explain the generation of the clusters (Situation 1: components 1 and 2 of the gaussian mixture generated cluster 1 and 2 respectively. Situation 2 (reverse): components 1 and 2 generated clusters 2 and 1 respectively).
+But since the clusters contain a different number of points, and the mixture components have different weights, one situation is less likely to be true than the other.
+This is highlighted by the fact that the top-left mode in the posterior shows a higher probability than the lower-right mode.
+
+
+
+## Single MCMC chain sampling
+We created a simple single MCMC chain for sampling this posterior distribution, defined in the module `single_chain.py`.
+
+### Sampling
+With an initial state for the MCMC chain set at (1, 0), the chain gets stuck on the lower probability mode.
+That means that the gaussian mixture component with the higher weight (component 2) fits the cluster with the lower number of points, and vice versa.
+
+<img alt="sampling single chain with initial state = (1, 0)" src="./images/sampling_sc_initstate-1-0.png" height="400"/>
+
+
+This results in the samples of the means being slightly offset from their expected positions (0 and 1.5), as to compensate for the mismatch.
+
+<img alt="sampled means single chain" src="./images/samples_means_sc_initstate-1-0.png" height="400"/>
+
+### Cluster assignment
+Soft k-means clustering treats the cluster assignments as probability distributions over the clusters.
+Each data point is associated with a log unnormalized probability distribution for each cluster, which is obtained from transformation of the samples of the means (called the "soft Z" transformed samples).
+These soft Z distributions, after softmax normalization, provide the probability of each point to belong to each cluster.
+The distributions of the soft Z samples for 3 individual data points are shown below (one point located on the left, one in the middle, and one on the right).
+
+<img alt="Soft_z analysis" src="./images/soft-z_samples_sc2.png"/>
```

Comment:
Can we remove the redundant "Point" in the subplot titles? Or, if we keep it, name the points, like Point 1, Point 2, Point 3 in the subplot titles. Or even just write x_!, x_2, x_3 :thinking: